src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,133 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int POSITION_FOR_NULL_KEYS = 0;
+    private int threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    private int capacity = INITIAL_CAPACITY;
+    private int size;
+    private Node<K, V>[] hashMap = new Node[INITIAL_CAPACITY];
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        if (key == null) {
+            putIfKeyNull(value);
+        } else {
+            putIfKeyNotNull(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
```

Comment:
this method is very complicate 
It can be simplified by taking index of a bucket and going though elements to get necessary element